packages/wix-ui-core/src/components/horizontal-menu/HorizontalMenu.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { TextAlignProperty, JustifyContentProperty } from 'csstype';
+
+import { HorizontalMenuItem } from './horizontal-menu-item';
+import { HorizontalMenuOption } from './horizontal-menu-option';
+import { HorizontalMenuColumnLayout } from './horizontal-menu-column-layout';
+import { HorizontalMenuGridLayout } from './horizontal-menu-grid-layout';
+import { HorizontalMenuListLayout } from './horizontal-menu-list-layout';
+
+import styles from './HorizontalMenu.st.css';
+
+export interface HorizontalMenuProps {
+  /** Menu container data-hook */
+  dataHook?: string;
+
+  /** Menu container width */
+  width?: string;
+
+  /** Render items with divider */
+  withDividers?: boolean;
+
+  /** Equal items width */
+  equalWidth?: boolean;
+
+  textAlign?: TextAlignProperty;
+
+  justifyContent?: JustifyContentProperty;
+}
+
+/** Horizontal menu */
+export class HorizontalMenu extends React.PureComponent<HorizontalMenuProps> {
+  static displayName = 'HorizontalMenu';
+
+  static Item = HorizontalMenuItem;
+
+  static Layout = {
+    Column: HorizontalMenuColumnLayout,
+    Grid: HorizontalMenuGridLayout,
+    List: HorizontalMenuListLayout,
+  };
+
+  static Option = HorizontalMenuOption;
+
+  static defaultProps = {
+    dataHook: 'horizontal-menu-container',
+    width: '100%',
+    withDividers: false,
+    equalWidth: true,
+    justifyContent: 'space-between',
+  };
+
+  render() {
+    const {
+      dataHook,
+      textAlign = 'left',
+      justifyContent,
+      width,
+      withDividers,
+      equalWidth,
+      children,
+    } = this.props;
+
+    const menuStyles = classNames(styles.menu, {
+      [styles.withDividers]: withDividers,
+      [styles.equalItems]: equalWidth,
+    });
+
+    return (
+      <nav
+        role="navigation"
+        {...styles('root', {}, this.props)}
+        data-hook="horizontal-menu-navigation"
+        style={{ width }}
```

Comment:
```suggestion
        style={this.props.style}
```

Maybe remove `width` property and pass style object from props.
In this way it will be possible to set width via style props along with any other style params.